src/custom.rs

```diff
@@ -45,21 +53,49 @@ pub enum ConfigLoadError {
     Invalid(#[from] serde_yaml::Error),
 }
 
-#[derive(Clone, Debug, Deserialize)]
+// TODO(JsonSchema): suggest the valid themes based on themes.presentation.theme_names() but allow
+// any string as a theme name
+#[derive(Clone, Debug, Deserialize, JsonSchema)]
 #[serde(deny_unknown_fields)]
 pub struct DefaultsConfig {
+    /// The theme to use by default in every presentation unless overridden.
+    #[schemars(schema_with = "theme_schema")]
     pub theme: Option<String>,
 
+    /// Override the terminal font size when in windows or when using sixel.
     #[serde(default = "default_font_size")]
+    #[validate(range(min = 1))]
     pub terminal_font_size: u8,
 
+    /// The image protocol to use.
     #[serde(default)]
     pub image_protocol: ImageProtocol,
 
+    /// Validate that the presentation does not overflow the terminal screen.
     #[serde(default)]
     pub validate_overflows: ValidateOverflows,
 }
 
+// allow the theme to be any string, but suggest the valid themes based on the themes
+fn theme_schema(_gen: &mut SchemaGenerator) -> Schema {
+    let mut schema = SchemaObject::default();
+    schema.metadata().description =
+        Some("The theme to use by default in every presentation unless overridden.".to_string());
+    schema.instance_type = Some(SingleOrVec::Vec(vec![schemars::schema::InstanceType::String]));
+    schema.enum_values = Some(vec![
```

Comment:
I know you spent some time to get this enum to work, but I'm not sure if it's the right way to go. Because you can have your own custom themes, this end up being misleading to users. I see that you do allow an arbitrary string as well, but I think if I see my autocompletion show a list of things (e.g. the build in themes), I would always assume I _need_ to use a value from that list so I may get confused if I have a local custom theme and it's not showing up.

Reply:
Sure, we can leave it out too. I'll revert it back to just a string with no suggestions for the built in themes.